Tools/clinic/clinic.py

```diff
@@ -2669,15 +2669,15 @@ class CConverter(metaclass=CConverterAutoRegister):
     # keep in sync with self_converter.__init__!
     def __init__(self,
              # Positional args:
-             name,
-             py_name,
+             name: str,
+             py_name: str,
              function,
              default=unspecified,
              *,  # Keyword only args:
-             c_default=None,
-             py_default=None,
-             annotation=unspecified,
-             unused=False,
+             c_default: str | None = None,
+             py_default: str | None = None,
+             annotation: str | Unspecified = unspecified,
```

Comment:
Doesn't need to be done now (and you may not like the suggestion), but you _could_ use an enum for all the sentinels that are defined in `clinic.py`:

```py
@enum.global_enum
class Sentinels(enum.Enum):
    UNSPECIFIED = 'Unspecified'
    NULL = 'Null'
    UNKNOWN = 'Unknown'

    def __repr__(self):
        return f'<{self.value}>'
```

If you did that, then you'd be able to use a `typing.Literal` annotation here, which would be more expressive of the fact that `unspecified` is a singleton:

```suggestion
             annotation: str | Literal[Sentinels.Unspecified] = unspecified,
```

Reply:
Neat!